Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -64,6 +66,44 @@ public FreezeOnMatchCustomization(Type targetType)
         }
 
         /// <summary>
+        /// Initializes a new instance of the <see cref="FreezeOnMatchCustomization"/> class.
+        /// </summary>
+        /// <param name="parameter">
+        /// The <see cref="ParameterInfo"/> used to freeze specimens decorated with advanced
+        /// attributes (e.g. <see cref="StringLengthAttribute"/>).
+        /// </param>
+        /// <param name="matcher">
+        /// The <see cref="IRequestSpecification"/> used to match the requests
+        /// that will be satisfied by the frozen specimen.
+        /// </param>
+        /// <exception cref="ArgumentNullException">
+        /// <paramref name="parameter"/> or <paramref name="matcher"/> is null.
+        /// </exception>
+        public FreezeOnMatchCustomization(
+            ParameterInfo parameter,
```

Comment:
Sorry for the late reply but I'm currently on vacation and have only occasional access to the Internet. ☀️🍦

Looking back at the discussion, I noticed that @ploeh did, in fact, [predict this exact scenario](https://github.com/AutoFixture/AutoFixture/pull/317#discussion_r19219817): 

> Based on this, I'm inclined to keep it as is. However, I was wondering if we'd then be missing out on some opportunities, like:

    new FreezeOnMatchCustomization(somePropertyInfo);
    new FreezeOnMatchCustomization(someParameterInfo);
    new FreezeOnMatchCustomization(someFieldInfo);
    // etc.

Unfortunately, I failed to follow up on that comment so the blame is totally on me. 😔

In the face of this new requirement, I agree that relaxing the constructor of `FreezeOnMatchCustomization` sounds lile the right thing to do.

Of course, we would still have to pay complexity tax of handling multiple variants of the `target` parameter. Although it would be nice if we could restrict the subset of types that can be passed as arguments through the type system, I'm afraid that, in the end, it would still have to be `object` since there is no other contravariant type between `PropertyInfo`, `ParameterInfo`, `FieldInfo` and `Type`.

I should mention that I'm saying all this without having researched other possible solutions that don't involve modifying the constructor of `FreezeOnMatchCustomization`.